components/brave_new_tab_ui/reducers/new_tab_reducer.ts

```diff
@@ -121,9 +121,14 @@ export const newTabReducer: Reducer<NewTab.State | undefined> = (state: NewTab.S
       }
       // Empty custom bg url means using brave background.
       const url = payload.customBackground.url.url
-      state.backgroundWallpaper =
-          url === '' ? backgroundAPI.randomBackgroundImage()
-                     : { wallpaperImageUrl: url }
+      const solidColor = payload.customBackground.solidColor
+      if (url !== '') {
+        state.backgroundWallpaper = { wallpaperImageUrl: url, wallpaperSolidColor: undefined }
```

Comment:
Nit: You can just omit the undefined keys

```suggestion
        state.backgroundWallpaper = { wallpaperImageUrl: url }
```